### DISCUSSION TITLE
Modal Analysis of Diamond Soft Robot

### DISCUSSION THREAD

**Discussion by jjalora** [SofaDefrost/SoftRobots#177] [Help - Using] [answered]
Hello,

I was conducting a modal analysis of the Diamond soft robot (i.e. solving the generalized eigenvalue problem $(M - \omega_n^2 K) u_n = 0$. I'm extracting the $K$ matrices as explained in [here](https://github.com/sofa-framework/sofa/discussions/2912) and introducing constraints as explained [here](https://github.com/sofa-framework/sofa/discussions/3134). 

I found the following modal displacements:
![image](https://user-images.githubusercontent.com/86177460/179640798-8f3f39c3-066c-4700-b727-b1fd4bd0aeed.png)

I've managed to obtain the same 3 dominant modes through a more roundabout way of extracting the $K$ matrix (prior to the direct extraction feature being released), so I'm starting to believe these are the true structural modes of the FEM. Intuitively, what I expected, though, is that the modes would be along the (xyz) axes of the robot given that its geometry is fairly symmetric. In other words, if I displace the robot along the $x$-axis and then let go, I expect that there would be no oscillation along the $y$-axis and vice versa. 

Is my intuition here wrong? Is there anyone with experience working with this particular robot who might be able to shed light on this?

Thanks for the help!
John

**Comment by olivier-goury** [comment, marked as answer]
Hello John!
That is very interesting. Indeed, your intuition is right I think. One reason for this unexpected behaviour could be that despite the fact the robot is symmetric, the FEM mesh is not and is also fairly coarse. I would expect that refining the mesh would lead to more confined modes along each axis.
Olivier

**Reply by jjalora** [reply]
Hi Olivier,

Ahhh this makes perfect sense. This is probably why I observe similar asymmetry in the dominant modes taken from PCA!

Thanks so much for this insight!
Best,
John
